src/postprocessor/compress.rs

```diff
@@ -15,9 +15,70 @@
 use super::Postprocessor;
 use crate::errors::Result;
 use simd_json::ValueAccess;
-use std::{io::Write, str};
+use std::{
+    io::Write,
+    str::{self, FromStr},
+};
 use tremor_script::Value;
 
+enum Algorithm {
+    Gzip,
+    Zlib,
+    Xz2,
+    Zstd,
+    Snappy,
+    Lz4,
+}
+
+impl FromStr for Algorithm {
+    type Err = crate::errors::Error;
+
+    fn from_str(s: &str) -> Result<Self> {
+        match s {
+            "gzip" => Ok(Algorithm::Gzip),
+            "zlib" => Ok(Algorithm::Zlib),
+            "xz2" => Ok(Algorithm::Xz2),
+            "snappy" => Ok(Algorithm::Snappy),
+            "lz4" => Ok(Algorithm::Lz4),
+            "zstd" => Ok(Algorithm::Zstd),
+            other => return Err(format!("Unknown compression algorithm: {other}").into()),
+        }
+    }
+}
+
+impl Algorithm {
+    pub fn into_postprocessor(self, config: Option<&Value>) -> Result<Box<dyn Postprocessor>> {
+        match config.get_i64("level") {
+            Some(compression_level) => match self {
+                Algorithm::Xz2 => match Xz2::with_config(compression_level) {
+                    Ok(codec) => Ok(codec),
+                    Err(msg) => Err(msg),
+                },
```

Comment:
I think you can remove the match here
```suggestion
                Algorithm::Xz2 => Xz2::with_config(compression_level),
```